### PR TITLE
Compile warnings

### PR DESCRIPTION
**Pull request by filiplazovic**
Changes include:
- Refactored Authority contract to avoid stack depth error
- Removed unused imports
- Fixed compile and solium warnings such as:
  - line length
  - formating of more than 5 function arguments
  - unused function arguments
  - etc.

There are solium warnings for experimental features, im not sure if its safe to remove these.
Solc warnings for external libraries will unfortunately stay: https://github.com/ethereum/solidity/issues/2675

### CODE REVIEW

File: contracts/PatriciaTree/PatriciaTreeProofs.sol [hunk 1]
```diff
@@ -5,14 +5,18 @@ pragma experimental "ABIEncoderV2";
...
 /// @notice More info at: https://github.com/chriseth/patricia-trie
 contract PatriciaTreeProofs {
   using Bits for uint;
   using Data for Data.Edge;
   using Data for Data.Label;
 
-  function getImpliedRoot(bytes key, bytes value, uint branchMask, bytes32[] siblings) public view returns (bytes32) { // solium-disable-line security/no-assign-params
+  function getImpliedRoot(bytes key, bytes value, uint branchMask, bytes32[] siblings) public // solium-disable-line security/no-assign-params
+  view
+  returns (bytes32)
```
Comment (elenadimitrova) [inline contracts/PatriciaTree/PatriciaTreeProofs.sol]: I understand we reach the line limit above but can we at least keep `view returns (bytes32)` on the same line?

File: contracts/ReputationMiningCycle.sol [hunk 2]
```diff
@@ -555,7 +577,11 @@ contract ReputationMiningCycle is PatriciaTreeProofs, DSMath {
     }
   }
 
-  function getExpectedSkillIdAndAddress( ReputationLogEntry storage logEntry, uint256 updateNumber ) internal returns (uint256 expectedSkillId, address expectedAddress) {
+  function getExpectedSkillIdAndAddress(ReputationLogEntry storage logEntry, uint256 updateNumber)
+  internal
+  view
```
Comment (elenadimitrova) [inline contracts/ReputationMiningCycle.sol]: Put `internal view` at the end of the function declaration line above. That's what we've followed everywhere else I think. The `returns..` can be on its own line to stand out

**Comment by elenadimitrova** [comment]
Sorry, forgot to mention in review, to check if we can disable `no-experimental` in `solium` checks to avoid these warnings

**Comment by filiplazovic** [comment]
I prefer suppressing these warnings by fixing them. In case where that is not possible i would leave them. But i guess we can supress them and be more cautious, no strong feelings

**Comment by elenadimitrova** [comment]
You are not sufficiently informed when you say this, no hard feelings :)
To understand whether we need `experimental pragmas` you need to understand what they do, so read the documentation first before you decide to "fix" them. I want you to come back and tell me what each of the two experimental pragmas does and whether and why we need it.

Then check https://github.com/duaraghav8/Solium/issues/201 where we've agreed this is a bad warning anyway that will be fixed in upcoming `solium`

**Comment by filiplazovic** [comment]
I agree that we should keep `experimental pragmas`, but my comment was about whether we should supress warnings (any warning, not just about `experimental pragmas`) by turning off the rule in config file.

You are right that i could do a research of what benefits of these experimental features give us. Looking from solidity releases i can see that `ABIEncoderV2` allows us to use structs and arbitrary length arrays as functions arguments and return values. And `v0.5.0` will disallow uninitialized storage pointers, old constructor sintax, `var` keywork etc.

**Comment by elenadimitrova** [comment]
Yes, I was not arguing that we disable the rule to simply silence the warning. I was saying we should silence the rule because 
a) we need those experimental pragmas so we can't remove like solium is asking 
b) that solium rule is just wrong and will be removed in the next release anyway

**Pull request merged**